src/buffer.cc

```diff
@@ -204,6 +205,82 @@ Result Buffer::CompareRMSE(Buffer* buffer, float tolerance) const {
   return {};
 }
 
+std::vector<double> Buffer::GetHistogramForChannel(uint32_t channel,
+                                                   uint32_t num_bins) const {
```

Comment:
I feel that `num_bins` has to be `256` for this function to make sense. I suggest adding an assert. You may also want to assert something about the `format_` (again) like `format_->IsUint8()`